example/bufferCopy/src/bufferCopy.cpp

```diff
@@ -21,6 +21,73 @@
 #include <cstdint>
 #include <iostream>
 
+namespace alpaka
+{
+    template<typename Pointer, int Dim>
+    struct accessor;
```

Comment:
Normally, `Pointer` would be just a `T*`. I think supporting `const T*` for readonly buffers is also a natural extension.

I could also see `Pointer` being `cl::sycl::global_ptr<T>`. But I think that could get us intro troubles with the casting for the 2D and 3D accessor.